chain/client/src/chunk_validation.rs

```diff
@@ -515,22 +545,25 @@ impl Client {
         let prev_chunk = self.chain.get_chunk(&prev_chunk_header.chunk_hash())?;
         let (main_state_transition, implicit_transitions, applied_receipts_hash) =
             self.collect_state_transition_data(&chunk_header, prev_chunk_header)?;
-        let witness = ChunkStateWitness {
-            chunk_header: chunk_header.clone(),
+        let witness_inner = ChunkStateWitnessInner::new(
```

Comment:
Hmm =\ now we lose the argument names. But I see why a new is needed. This sucks but oh well.

Reply:
Maybe that's where Rust macroses are useful, if they allow to add a special field (signature) which is automatically set on creation. But that can wait